### PR TITLE
Add analysis task to check for waiting pods

### PR DESCRIPTION
**Pull request by philbrookes**
# Motivation
The analysis tasks in the dump tool should report any pods with waiting containers.

# Changes
Add an analysis task to check for waiting containers and report them

Add unit tests for the next check.

# Jiras
[RHMAP-9055](https://issues.jboss.org/browse/RHMAP-9055)

### CODE REVIEW

File: analysis.go [hunk 1]
```diff
@@ -128,6 +148,31 @@ func getDumpedJSONResourceFactory(basepath []string) DumpedJSONResourceFactory {
...
+func CheckForWaitingPods(JSONResourceFactory DumpedJSONResourceFactory) (Result, error) {
+	result := Result{Status: 0, StatusMessage: "this issue was not detected", CheckName: "check pods for 'waiting' containers", Info: []Info{}, Events: []Event{}}
+	pods := Pods{}
+	if err := JSONResourceFactory([]string{"definitions", "pods.json"}, &pods); err != nil {
+		result.Status = 2
+		result.StatusMessage = "Error executing task"
+		return result, err
+	}
+
+	for _, pod := range pods.Items {
```
Comment (maleck13) [inline analysis.go]: pretty sure that the slice will be initialised even if  there are no items , but will pod.Status.ContainerStatuses be initialised / have a value or could you hit a nil pointer here?

File: analysis_test.go [hunk 2]
```diff
@@ -151,42 +193,98 @@ func MockDeployConfigWithReplicaZero(p []string, d interface{}) error {
...
+		]
+	}`
+
+	decoder := json.NewDecoder(bytes.NewBuffer([]byte(contents)))
+	err := decoder.Decode(&d)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```
Comment (maleck13) [inline analysis_test.go]: could it be the case that the pods json is empty? If so might be worth adding test for this.

File: analysis.go [hunk 3]
```diff
@@ -128,6 +148,31 @@ func getDumpedJSONResourceFactory(basepath []string) DumpedJSONResourceFactory {
 
 }
 
+// CheckForWaitingPods checks all pods for any containers in waiting status
+func CheckForWaitingPods(JSONResourceFactory DumpedJSONResourceFactory) (Result, error) {
+	result := Result{Status: 0, StatusMessage: "this issue was not detected", CheckName: "check pods for 'waiting' containers", Info: []Info{}, Events: []Event{}}
+	pods := Pods{}
+	if err := JSONResourceFactory([]string{"definitions", "pods.json"}, &pods); err != nil {
+		result.Status = 2
+		result.StatusMessage = "Error executing task"
```
Comment (maleck13) [inline analysis.go]: should the error message have err.Error() appended?

**Comment by maleck13** [comment]
reviewed

**Comment by philbrookes** [comment]
Discussed on bluejeans, resolved comments.

**Comment by pmccarthy** [comment]
For validation I ran this against the OSM1 cluster which has multiple containers in a Waiting state. The resulting analysis.json file detected these containers, see below:

```
{
            "checkName": "check pods for 'waiting' containers",
            "status": 1,
            "statusMessage": "Waiting containers have been detected",
            "info": [
                {
                    "Name": "fh-mbaas",
                    "Namespace": "phils-node4-mbaas3",
                    "Kind": "container",
                    "Count": 1,
                    "Message": "container fh-mbaas in pod fh-mbaas-3-3k4m2 is in waiting state"
                }
            ],
            "events": []
        }
```
In terms of the Golang code, @philbrookes @maleck13 I'll leave that to you guys :)

**Pull request merged**